app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -73,6 +86,38 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
         }
     }
 
+    override fun spawnCurrencyModal() {
+        currencyDialog = AlertDialog.Builder(this)
+        val currencyDialogView = layoutInflater.inflate(R.layout.currency_modal, null)
+        currencyDialog!!.setView(currencyDialogView)
+        currencyDialog!!.setCancelable(true)
+        var custom_dialog = currencyDialog!!.create()
+        custom_dialog.show()
+        var convert_from = "CAD"
+        var convert_to = "CAD"
+
+        custom_dialog.findViewById<Spinner>(R.id.convert_from).onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                convert_from = custom_dialog.findViewById<Spinner>(R.id.convert_from).getSelectedItem().toString()
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {}
+        }
+
+        custom_dialog.findViewById<Spinner>(R.id.convert_to).onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                convert_to = custom_dialog.findViewById<Spinner>(R.id.convert_to).getSelectedItem().toString()
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {}
```

Comment:
Literally the same function from line 104

Reply:
We are defining a new listener for the second spinner, it is mandatory to implement this method for a non-abstract object.